go/ics23.go

```diff
@@ -36,11 +36,19 @@ type CommitmentRoot []byte
 func VerifyMembership(spec *ProofSpec, root CommitmentRoot, proof *CommitmentProof, key []byte, value []byte) bool {
 	// decompress it before running code (no-op if not compressed)
 	proof = Decompress(proof)
-	ep := getExistProofForKey(proof, key)
+	comparedKey, err := DoHashOrNoop(spec.GetPrehashComparedKey(), key)
+	if err != nil {
+		return false
+	}
+	comparedValue, err := DoHashOrNoop(spec.GetPrehashComparedValue(), value)
+	if err != nil {
+		return false
```

Comment:
Can you explain a use case of hashing the preimage value? Seems like it wouldn't be very practical as that means the tree doesn't store the preimage of the value and since hash functions are non-reversible, it'd need to be stored somewhere else anyways for an application to make use of the value. Thoughts?

Reply:
It seems to me we should just leave this out and revisit once there is an actual user for this field

Reply:
In our case, the tree (Sparse Merkle Tree) is stored on memory and has hashed values with a fixed length. It only checks the brief existence (checks whether we need to read the original value on the disk) and gets the proofs.